### ISSUE TITLE
Copy and Paste Test Steps, Properties and Data Table Rows

### ISSUE THREAD

**Issue by simonalexander2005** [zheng-wang/irontest#24]
When I have many test cases that use similar test steps, it would be very helpful to be able to copy/paste them, instead of having to recreate them along with all the assertions each time. Likewise for data table rows, where I might have the same test data setup for different tests

**Comment by zheng-wang** [comment]
Please have a try on the Test Case copy paste. I have been using this for long time. I have a folder which contains 'template' test cases for all kinds of 'patterns'. Each 'template' test case contains pre-created test steps which use managed endpoints and properties.

When I create a new test case, I just copy from a 'template' test case, and then tailor the new test case by populating few details specific to it, like input message and expected output message. I use properties in my 'templates', so most of the time I only need to populate data on the Properties tab of the new test case, and then run it. This did save me a lot of time.

My experience is that copy and paste at test steps, properties and data table rows levels is actually less efficient than copy and paste at test case level.

**Comment by simonalexander2005** [comment]
I am setting up some test cases which need some steps from one test case, and some from another; so currently I can copy one of the test cases but then need to re-create half of the test steps that come from the other test case; rather than just being able to copy and paste them

**Comment by zheng-wang** [comment]
I suggest you create many 'template' test cases, and organize them in subfolders, so that most of times you can just copy test case, and don't need to care about test steps. If you find yourself working on test steps a lot, just create more 'template' test cases so as to avoid it. I believe you will feel a lot relieved when getting there.

**Comment by zheng-wang** [comment]
On the other hand, copy/paste at Test Steps, Properties and Data Table Rows levels is definitely useful, but it shouldn't be used often. Otherwise it is going to make user experience close to using code.

Iron Test is designed to relieve user from reading too many details. You can achieve everything by using Java code (JUnit), but the effort is to read a lot of code again and again every day. Reading code is brain energy burning.

Similarly, reading and digging into test steps details is also brain energy burning (though less than reading and digging into code). Iron Test intends to help user dump details out of mind (and into reusable test cases), and avoid user remembering and repeating so many details so frequently.

**Comment by simonalexander2005** [comment]
Yes, I think we agree about the aim - it's the initial set up where the copy-and-pasting would be useful

**Comment by zheng-wang** [comment]
Ok thanks, the feature makes the tool's copy/paste capability more complete, and especially useful when creating new 'template' test cases from existing 'template' or 'populated' test cases. But due to not used often in the long term, I'm leaving it as a low priority item to implement.
